### PR TITLE
[sleep mode] clear pytorch cache after sleep

### PR DESCRIPTION
**Pull request by lionelvillard**
Clear PyTorch cache right after putting the instance to sleep.

For ibm-granite/granite-20b-code-instruct-8k, memory use before this fix is:

```
INFO 03-20 19:40:47 [gpu_worker.py:81] Sleep mode freed 37.55 GiB memory, 3.29 GiB memory is still in use.
```

And with this fix:

```
INFO 03-20 19:45:25 [gpu_worker.py:81] Sleep mode freed 39.48 GiB memory, 1.36 GiB memory is still in use.
```

### CODE REVIEW

**Review by maxdebayser — COMMENTED** [review]
Intuitively this makes total sense. But I don't know if there could be any side effects.

**Comment by njhill** [comment]
Also LGTM but would also be good to get a nod from @youkaichao.

**Comment by youkaichao** [comment]
i think you are clearing some buffers for sampling logits. it works, but people have to be careful, because when you wake up the engine, and serve the requests, samplers will allocate that part of memory again.

**Review by youkaichao — APPROVED** [review]
LGTM if you want to reclaim as much memory as possible when sleeping.

**Comment by lionelvillard** [comment]
Can we merge this PR?
